project_from_quotation/models/sale_order.py

```diff
@@ -0,0 +1,193 @@
+        """
+        return {
+            "name": sale_order.name,
+            "quotation_id": sale_order.id,
+            "active": True,
+        }
+
+    def _get_order_lines_unique_products(self, order_lines=None):
+        """
+        Get order lines with unique products
+
+        Args:
+            order_lines (sale.order.line recordset): Order lines
+            if not set, uses self.order_line
+
+        Returns:
+            sale.order.line recordset: Every first
+            line with the same product
+        """
+        self.ensure_one()
+        unique_products = []
+        res_order_lines = self.env["sale.order.line"]
+        order_lines = (order_lines or self.order_line).filtered("product_id")
+
+        for line in order_lines:
+            if line.product_id.id not in unique_products:
+                unique_products.append(line.product_id.id)
+                res_order_lines |= line
+
+        return res_order_lines
+
+    def _create_project_from_quotation(
+        self,
+        mode="project",
+        order_lines=None,
+    ):
+        """
+        Create project from a quotation
+
+        Args:
+        mode (str): Project creation mode. Possible values:
+            project - create a new empty project
+            tasks_per_line - create a project with tasks for each order line
+            tasks_per_product - create a project with tasks for each product
+        order_lines (sale.order.line): Order lines to use for tasks creation
+
+        Returns:
+            Created record of the project.project model
+        """
+        self.ensure_one()
+        project = False
+        task_obj = self.env["project.task"]
+        project_obj = self.env["project.project"]
+        order_lines = (order_lines or self.order_line).filtered(lambda x: not x.task_id)
+
+        project_template = project_obj.browse(
+            int(
+                self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("project_from_quotation.template_project_id")
+            )
+        )
+        if project_template:
+            project = project_template.copy(self._prepare_project_vals(self))
+        else:
+            project = project_obj.create(self._prepare_project_vals(self))
+
+        if mode.startswith("tasks"):
+            tasks = self.env["project.task"]
+            if mode == "tasks_per_line":
+                task_vals = [
+                    self._prepare_task_vals(project, line) for line in order_lines
+                ]
+            elif mode == "tasks_per_product":
+                task_vals = [
+                    self._prepare_task_vals(project, line)
+                    for line in self._get_order_lines_unique_products(order_lines)
+                ]
+            template_task = task_obj.browse(
+                int(
+                    self.env["ir.config_parameter"]
+                    .sudo()
+                    .get_param("project_from_quotation.template_task_id")
```

Comment:
What if `project_from_quotation.template_task_id` is not defined? Would'n it raise an exception because of trying to perform `int(None)`?